website/docs/opengl/c1/2-hello-quad.md

```diff
@@ -0,0 +1,524 @@
+---
+{
+    "TableOfContents": {
+        "Name": "1.2 - Hello Quad",
+        "Url": "2-hello-quad.html",
+        "Metadata": {
+            "AuthorGitHub": "ohtrobinson",
+            "DateTimeWritten": "02/11/2022 12:00",
+            "PreviewImage": ""
+        }
+    }
+}
+---
+
+# 1.2 - Hello Quad
+<?# Info "You can view the source code for this tutorial [here.](../sources/1.2-final-result.html) This tutorial builds on the previous tutorial. If you haven't read it, you can do so [here.](1-hello-window.html)" /?>
+
+Let's draw something on-screen! In this tutorial, you'll learn:
+
+* How to initialize OpenGL.
+* How to clear the window.
+* How the OpenGL pipeline works.
+* What vertex array objects, vertex buffers, and element array buffers are, and how to use them.
+* How to create and use shaders.
+* How to draw to the screen.
+
+This tutorial will feature a lot of content and new info, so we'll take it slow and with lots of explanations.
+
+## Initializing OpenGL
+The first thing we need to do before we can issue OpenGL commands, is to initialize it. Fortunately, Silk.NET windowing does most of the hard part for us, and we only have to get the GL api.
+
+Add the following using directive to the top of your file, you'll need this to use OpenGL:
+
+```cs
+using Silk.NET.OpenGL;
+```
+
+Then, at the top of your class, add the following:
+
+```cs
+private static GL _gl;
+```
+
+Finally, in your `OnLoad` method, add the following:
+
+```cs
+_gl = _window.CreateOpenGL();
+```
+
+What are we doing here? Silk.NET requires you to keep a **reference** to the OpenGL API. If you've used or seen OpenGL in C, you'll notice that this is different to the way that it is done there. This is done so that you can more easily keep track of multiple contexts. If you don't know what that is, don't worry about it for now, we won't be using it in these tutorials.
+
+<?# Info "If you take a look at the source in the Silk.NET samples repository, you will notice that it uses `_gl = GL.GetApi(_window)`. This is another way to get the GL API, however when using Silk.NET windowing, it's recommended that you use `_window.CreateOpenGL()` instead." /?>
+
+Now, run your application again. If all is good, you should see no change. Awesome! Let's do our first steps in OpenGL: Clearing the window.
+
+## Clearing the window
+Before we start this, let's take a look at what makes up an OpenGL window. 
+
+A window consists of several **framebuffers**. A framebuffer is essentially a texture that can be drawn to by OpenGL. An OpenGL window will consist of the following framebuffers:
+
+* Color buffer
+* Depth stencil buffer
+
```

Comment:
```suggestion
A window consists of several default **framebuffers**. A framebuffer is a set of buffers that can be rendered to, essentially a texture that can be drawn to by OpenGL. An OpenGL window's default framebuffers will consist of the following buffers:

* Color buffer
* Depth stencil buffer
```

According to [the khronos docs](https://www.khronos.org/opengl/wiki/Framebuffer):
> A Framebuffer is a collection of buffers that can be used as the destination for rendering.

So a framebuffer _contains_ the color, depth and stencil buffers. Still, let's keep the "essentially a texture that can be drawn to" because that's much easier to understand for newbies :)

Reply:
I agree about changing it to a "set of buffers", that is an oversight on my part. However I dislike calling them "default framebuffers", since, while yes the window has a default framebuffer, I do mention (i think?) that you can't directly interact with the window's framebuffer. And for example in Direct3D you *can* interact with the window's framebuffer, I prefer to say that the window consists of several framebuffers instead of *default* framebuffer. (I could go on with a long rant about the fact that you can't directly interact with the window's framebuffer... just give me the damn framebuffer so I can mess with it...!)

Reply:
Fair enough lol. _Damn you, window framebuffers!_